src/python/WMCore/MicroService/Unified/MSRuleCleaner.py

```diff
@@ -396,12 +396,16 @@ def getRucioRules(self, wflow, gran, rucioAcct):
         # Find all the output placement rules created by the agents
         for dataCont in wflow['OutputDatasets']:
             if gran == 'container':
-                wflow['RulesToClean'][currPline].extend(self.rucio.listDataRules(dataCont, account=rucioAcct))
+                ruleIds = [rule['id'] for rule in
```

Comment:
I guess you already know I'm not a big fan of multi-lines list comprehension, so let me suggest a construction that looks clearer in my opinion. How about:
```
for rule in self.rucio.listDataRules(dataCont, account=rucioAcct):
    wflow['RulesToClean'][currPline].append(rule['id'])
```
?

If you agree, please change the block-level code (below) as well

Reply:
Done